SoftLayer/managers/hardware.py

```diff
@@ -467,13 +467,18 @@ def _generate_create_dict(self,
             'hostname': hostname,
             'domain': domain,
         }
-
+        try:
+            size = _get_preset_id(package, size)
+        except SoftLayer.SoftLayerError as err:
+            if not err.message.startswith('Could not find valid size for'):
+                raise
+            size = size
```

Comment:
As you asked I check if a size exists with its name before take ID.

I check with try/except, and raise other exceptions.